src/matchers/tables.lua

```diff
@@ -1,13 +1,124 @@
+local prettyValue = require("src.utils.prettyValue")
+local deepEqual = require("src.utils.deepEqual")
+local assertType = require("src.asserts.type")
+
 ---@type lest.Matcher
-local function toHaveLength(ctx, received, length) end
+local function toHaveLength(ctx, received, length)
+	assertType(length, "number")
+
+	return {
+		pass = #received == length,
+		message = string.format(
+			"Expected %s to%shave the length of %d",
+			prettyValue(received),
+			ctx.inverted and " not " or " ",
+			length
+		),
+	}
+end
 
 ---@type lest.Matcher
-local function toContain(ctx, received, item) end
+local function toContain(ctx, received, item)
+	local pass = false
+
+	if type(received) == "string" then
+		assertType(item, "string")
+		pass = received:find(item) ~= nil
+	elseif type(received) == "table" then
+		for _, value in pairs(received) do
+			if value == item then
+				pass = true
+				break
+			end
+		end
+	else
+		error("Expected to receive a string or a table")
```

Comment:
So I just omitted this since it will implicitly fail the matcher because `pass` is initialized to `false`. I thought it could be more explicit but didn't want to complicate it further. This line is now gone, which fails the matcher instead of erroring now, albeit less clearly, let me know if that's ok.